src/dashboard/src/components/advanced_search.py

```diff
@@ -167,8 +197,17 @@ def query_clause(index, queries, ops, fields, types):
             else:
                 term_field = '_all'
             return {'term': {term_field: queries[index]}}
-    else:
+    elif types[index] == 'string':
         return {'query_string': {'query': queries[index], 'fields': search_fields}}
+    elif types[index] == 'range':
+        start, end = _parse_date_range(queries[index])
+        for date in (start, end):
+            try:
+                datetime.strptime(date, '%Y-%m-%d')
```

Comment:
If you're parsing user-inputted dates, [dateutil.parse](http://labix.org/python-dateutil#head-c0e81a473b647dfa787dc11e8c69557ec2c3ecd2) may be useful.  It does fuzzy matching.